src/config.rs

```diff
@@ -402,17 +437,140 @@ mod tests {
+
+    fn minimal_config() -> ConfigFile {
+        let rootfs = Rootfs {
+            r#type: "layers".into(),
+            diff_ids: vec![
+                "sha256:c6f988f4874bb0add23a778f753c65efe992244e148a1d2ec2a8b664fb66bbd1".into(),
+                "sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef".into(),
+            ],
+        };
+
+        ConfigFile {
+            architecture: Architecture::Amd64,
+            os: Os::Linux,
+            config: None,
+            rootfs,
+            history: None,
+            created: None,
+            author: None,
+        }
+    }
+
+    const MINIMAL_CONFIG2: &str = r#"
+    {
+        "architecture":"arm64",
+        "config":{
+            "Env":["PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"],
+            "WorkingDir":"/"
+        },
+        "created":"2023-04-21T11:53:28.176613804Z",
+        "history":[{
+            "created":"2023-04-21T11:53:28.176613804Z",
+            "created_by":"COPY ./src/main.rs / # buildkit",
+            "comment":"buildkit.dockerfile.v0"
+        }],
+        "os":"linux",
+        "rootfs":{
+            "type":"layers",
+            "diff_ids":[
+                "sha256:267fbf1f5a9377e40a2dc65b355000111e000a35ac77f7b19a59f587d4dd778e"
+            ]
+        }
+    }"#;
+
+    fn minimal_config2() -> ConfigFile {
+        let config = Some(Config {
+            env: Some(vec![
+                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin".into(),
+            ]),
+            working_dir: Some("/".into()),
+            ..Config::default()
+        });
+        let history = Some(vec![History {
+            created: Some(
+                Utc.datetime_from_str("2023-04-21T11:53:28.176613804Z", "%+")
+                    .expect("parse time failed"),
+            ),
+            author: None,
+            created_by: Some("COPY ./src/main.rs / # buildkit".into()),
+            comment: Some("buildkit.dockerfile.v0".into()),
+            empty_layer: None,
+        }]);
+
+        let rootfs = Rootfs {
+            r#type: "layers".into(),
+            diff_ids: vec![
+                "sha256:267fbf1f5a9377e40a2dc65b355000111e000a35ac77f7b19a59f587d4dd778e".into(),
+            ],
+        };
+
+        ConfigFile {
+            architecture: Architecture::Arm64,
+            os: Os::Linux,
+            config,
+            rootfs,
+            history,
+            created: Some(
+                Utc.datetime_from_str("2023-04-21T11:53:28.176613804Z", "%+")
+                    .expect("parse time failed"),
+            ),
+            author: None,
+        }
+    }
+
     #[test]
-    fn deserialize() {
+    fn deserialize_example() {
         let example = example_config();
         let parsed: ConfigFile = serde_json::from_str(EXAMPLE_CONFIG).expect("parsed failed");
         assert_eq!(example, parsed);
     }
 
     #[test]
-    fn serialize() {
+    fn deserialize_minimal() {
```

Comment:
I think it would be nice to remove all this repetition by using the [`rstest`](https://crates.io/crates/rstest) library, like we are doing in other parts of the code.

I would propose to consolidate all these tests into a single one, something like:
```rust
#[rstest]
#[case(0, 0)]
#[case(1, 1)]
#[case(2, 1)]
#[case(3, 2)]
#[case(4, 3)]
fn fibonacci_test(#[case] name, #[case] input: &str, #[case] expected: ConfigFile) {
        let parsed: ConfigFile = serde_json::from_str(input).expect(format!("{name}: parsed failed").as_str());
        assert_eq!(expected, parsed, "something went wrong with test case {name}");
}
```